docs/README.md

```diff
@@ -0,0 +1,43 @@
+## Kubernetes-Mesos Documentation
+
+This project combines concepts and technologies from two already-complex projects: Mesos and Kubernetes.
+It may help to familiarize yourself with the basics of each project before reading on:
+
+* [Mesos Documentation][1]
+* [Kubernetes Documentation][2] and [Key Concepts][4]
+  * [Labels][6]
+  * [Nodes][9]
+  * [Pods][7]
+  * [Replication Controller][8]
+  * [Services][5]
+
+It may also be helpful to understand the [known issues][3] with respect to this project.
+
+### Architecture
+
+A Mesos cluster consists of one or more masters, and one or more slaves.
+Kubernetes-Mesos (k8sm) operates as a Mesos framework that runs on the cluster.
+As a framework, k8sm provides scheduler and executor components, both of which are hybrids of Kubernetes and Mesos:
+the scheduler component integrates the Kubernetes scheduling API and the Mesos scheduler runtime, whereas;
+the executor component integrates Kubernetes kubelet services and the Mesos executor runtime.
+
+Multiple Mesos masters are typically configured to coordinate leadership election via Zookeeper.
+Kubernetes maintains its internal registry (pods, replication controllers, bindings, minions, services) in etcd.
+Users typically interact with Kubernetes using the `kubecfg` or `kubectl` commands to manage Kubernetes primitives.
+
+When a pod is created in Kubernetes, the k8sm scheduler creates an associated Task and queues it for scheduling.
+Upon pairing the pod/task with an acceptable resource offer, the scheduler binds the pod/task to the offer's slave.
+As a result of binding the pod/task is launched and delivered to an executor (an executor is created by the Mesos slave if one is not already running).
+The executor launches the pod/task, which registers the bound pod with the kubelet engine and the kubelet begins to manage the lifecycle of the pod instance.
+
+![Architecture Diagram](architecture.png)
```

Comment:
By all means, please have the designers fix it up.

On Tue, Jan 27, 2015 at 2:43 AM, adam-mesos <notifications@github.com>
wrote:

> In docs/README.md
> <https://github.com/mesosphere/kubernetes-mesos/pull/134#discussion_r23591432>
> :
>
> > +A Mesos cluster consists of one or more masters, and one or more slaves.
> > +Kubernetes-Mesos (k8sm) operates as a Mesos framework that runs on the cluster.
> > +As a framework, k8sm provides scheduler and executor components, both of which are hybrids of Kubernetes and Mesos:
> > +the scheduler component integrates the Kubernetes scheduling API and the Mesos scheduler runtime, whereas;
> > +the executor component integrates Kubernetes kubelet services and the Mesos executor runtime.
> > +
> > +Multiple Mesos masters are typically configured to coordinate leadership election via Zookeeper.
> > +Kubernetes maintains its internal registry (pods, replication controllers, bindings, minions, services) in etcd.
> > +Users typically interact with Kubernetes using the `kubecfg` or `kubectl` commands to manage Kubernetes primitives.
> > +
> > +When a pod is created in Kubernetes, the k8sm scheduler creates an associated Task and queues it for scheduling.
> > +Upon pairing the pod/task with an acceptable resource offer, the scheduler binds the pod/task to the offer's slave.
> > +As a result of binding the pod/task is launched and delivered to an executor (an executor is created by the Mesos slave if one is not already running).
> > +The executor launches the pod/task, which registers the bound pod with the kubelet engine and the kubelet begins to manage the lifecycle of the pod instance.
> > +
> > +![Architecture Diagram](architecture.png)
>
> Love the diagram and description, but I bet our designers can come up with
> better colors/fonts. Mind if they take a stab?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mesosphere/kubernetes-mesos/pull/134/files#r23591432>.
>



-- 
James DeFelice
585.241.9488 (voice)
650.649.6071 (fax)